docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set.md

```diff
@@ -0,0 +1,163 @@
+
+If you find existing PetSets and you have all their manifests at hand, you can continue to the next step to prepare StatefulSet manifests. 
+
+Otherwise, you need to save their manifests so that you can recreate them as StatefulSets later. 
+Here's an example command for you to save all existing PetSets as one file. 
+
+```shell
+# Save all existing PetSets in all namespaces into a single file. Only needed when you don't have their manifests at hand. 
+kubectl get petsets --all-namespaces -o yaml > all-petsets.yaml
+```
+
+#### Prepare StatefulSet manifests 
+
+Now, for every PetSet manifest you have, prepare a corresponding StatefulSet manifest with the following steps: 
+
+1. Change `apiVersion` from `apps/v1alpha1` to `apps/v1beta1`
+2. Change `kind` from `PetSet` to `StatefulSet`
+3. If you have the debug hook annotation `pod.alpha.kubernetes.io/initialized` set to `true`, you can remove it because it's redundant. If you don't have this annotation, you should add one, with the value set to `false`, to pause StatefulSets operations.
+
+It's recommended that you keep both PetSet manifests and StatefulSet manifests, so that you can safely roll back and recreate your PetSets, 
+if you decide not to upgrade your cluster. 
+
+#### Delete all PetSets without cascading
+
+If you find existing PetSets in your cluster in the previous step, you need to delete all PetSets *without cascading*. You can do this from `kubectl` with `--cascade=false`. 
+Note that if the flag isn't set, **cascading deletion will be performed by default**, and all Pods managed by your PetSets will be gone. 
+
+Delete those PetSets by specifying file names. This only works when
+the files contain only PetSets, but not other resources such as Services:
+
+```shell
+# Delete all existing PetSets without cascading 
+# Note that <pet-set-file> should only contain PetSets that you want to delete, but not any other resources
+kubectl delete -f <pet-set-file> --cascade=false
+```
+
+Alternatively, delete them by specifying resource names: 
+
+```shell
+# Alternatively, delete them by name and namespace without cascading
+kubectl delete petsets <pet-set-name> -n=<pet-set-namespace> --cascade=false
+```
+
+Make sure you've deleted all PetSets in the system:
+
+```shell
+# Get all PetSets again to make sure you deleted them all 
+# This should return nothing
+kubectl get petsets --all-namespaces
+```
+
+At this moment, you've deleted all PetSets in your cluster, but not their Pods, Persistent Volumes, or Persistent Volume Claims. 
+However, since the Pods are not managed by PetSets anymore, they will be vulnerable to node failures until you finish the master upgrade and recreate StatefulSets.
+
+#### Upgrade master to Kubernetes version 1.5 or later
+
+Now, you can [upgrade your Kubernetes master](/docs/admin/cluster-management/#upgrading-a-cluster) to Kubernetes version 1.5 or later.
+Note that Kubernetes nodes should NOT be upgraded at this point, since the Pods
```

Comment:
This is pretty important, so maybe bold it. Also active voice will help to clarify and emphasize this point:

```
Note that **you should not upgrade Nodes at this time**, because ...
```